lua/barbar/state.lua

```diff
@@ -296,5 +324,41 @@ function state.icons(bufnr, activity)
   return buffer_icons
 end
 
+-- Save/load state
+
+--- Save recently_closed list
+--- @return nil
+function state.save_recently_closed()
+  local filepath = stdpath('cache') .. '/barbar.json'
+  local file = io.open(filepath, 'w+')
+  if file == nil then
+    return
+  end
+
+  local encoded_state = json_encode({
+    recently_closed = state.recently_closed,
+  })
+
+  file:write(encoded_state)
+  file:close()
```

Comment:
```suggestion
  local filepath = stdpath('cache') .. '/barbar.json'
  local file, open_err = io.open(filepath, 'w')
  if open_err ~= nil then
    return utils.notify(err, vim.log.levels.ERROR)
  elseif file == nil then
    return utils.notify('Could not open ' .. filepath, vim.log.levels.ERROR)
  end

  do
    local _, write_err = file:write(json_encode {recently_closed = state.recently_closed})
    if write_err ~= nil then
      return utils.notify(write_err, vim.log.levels.ERROR)
    end
  end

  local success, close_err = file:close()
  if close_err ~= nil then
    return utils.notify(err, vim.log.levels.ERROR)
  elseif success == false then
    return utils.notify('Could not close ' .. filepath, vim.log.levels.ERROR)
  end
```

We weren't checking for errors (e.g. wrong permissions, `filepath` doesn't exist on disk, etc.)